rats-tls/src/core/cpu.c

```diff
@@ -9,23 +9,131 @@
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef SGX
+#include "rtls_t.h"
+#else
 #include <sys/stat.h>
 #include <sys/sysmacros.h>
+#endif
+
+#ifdef SGX
+static inline void __cpuidex(int a[4], int b, int c)
+{
+	a[0] = b;
+	a[2] = c;
+	ocall_cpuid(&a[0], &a[1], &a[2], &a[3]);
+}
+
+static bool is_legacy_oot_kernel_driver(void)
+{
+	bool retval;
+	int dev_num = SGX_DEVICE_MAJOR_NUM;
+
+	ocall_is_device(&retval, "/dev/isgx", &dev_num);
```

Comment:
You should implement ocall_stat() instead of ocall_is_device(). ocall_stat is more general and useful with the extension. In addition, the check to see whether dev_num is SGX_DEVICE_MAJOR_NUM should occur in trusted side rather than untrusted side.

At last, please move all sgx mode codes **after** the general one, e.g:
```C
# ifndef SGX
// ....
# else
// ....
# endif
```

Reply:
For #1: 
Your suggestion is right ideally. There is consideration for ocall_stat because "struct stat" can't be input parameter in ocall compiling rule. Let me find a compromise way for ocall_stat.
For #2:
Currently alignment format is set be .clang-format. Do you mean change all "#ifdef SGX" format to "# ifdef SGX"? 
If yes, we can file another issue to track this request.

Reply:
@zhiminghufighting For #2, I mean putting the new codes for sgx mode to '# else' branch. Let the non-sgx mode code ahead of new codes.

Reply:
@zhiminghufighting OK well. So the current logic can be simplified as
```C
static bool is_legacy_oot_kernel_driver(void)
{
	return ocall_is_sgx_dev(&retval, "/dev/isgx");
}

bool ocall_is_sgx_dev(const char *dev)
{
	struct stat st;

	if (!stat(dev, &st))
	    return false;
		
	
	return (S_ISCHR(st.st_mode) && (major(st.st_rdev) == SGX_DEVICE_MAJOR_NUM);
}
```

Reply:
make sense

Reply:
By the way, the return type "bool" will be reconstructed by sgx sdk rule as such type after compilation:
sgx status SGX_ECDL ocall_is_sgx_dev(bool *retval, const char *dev)
So let's keep this function in  the format "ocall_is_sgx_dev(bool *retval, const char *dev)" will be more clear in definition.

Reply:
> 

Looks perfect.